src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneIndexInitializerService.cs

```diff
@@ -11,42 +10,32 @@ namespace OrchardCore.Lucene
     public class LuceneIndexInitializerService : IModularTenantEvents
     {
         private readonly ShellSettings _shellSettings;
-        private readonly ILogger _logger;
-        private readonly LuceneIndexSettingsService _luceneIndexSettingsService;
-        private readonly LuceneIndexingService _luceneIndexingService;
 
-        public LuceneIndexInitializerService(
-            LuceneIndexSettingsService luceneIndexSettingsService,
-            LuceneIndexingService luceneIndexingService,
-            ShellSettings shellSettings,
-            ILogger<LuceneIndexInitializerService> logger)
+        public LuceneIndexInitializerService(ShellSettings shellSettings)
         {
-            _luceneIndexSettingsService = luceneIndexSettingsService;
-            _luceneIndexingService = luceneIndexingService;
             _shellSettings = shellSettings;
-            _logger = logger;
         }
 
         public Task ActivatedAsync()
         {
-            if (_shellSettings.State != TenantState.Uninitialized)
+            if (_shellSettings.State == TenantState.Running)
```

Comment:
Positivism :)

Reply:
@Skrypt 

Here i suggested to @sebastienros to use `!= TenantState.Uninitialized` meaning not during a setup

But `== TenantState.Running` is okay, knowing that the other states are `Disabled` or `Invalid`